### PR TITLE
[Windows] Implement minor improvements, rev 2

### PR DESCRIPTION
**Pull request by maxim-lobanov**
# Description
This PR brings minor improvements to Windows image generation:
1) Make sure that `MSYS2` installation works correctly if no mingw, msys packages are specified in toolset
2) `Install-WDK.ps1` and `Install-Wix.ps1`  should fail on different platforms because they are not compatible with them
3) Don't run TLS tests in scope of `images/win/scripts/Installers/Update-DotnetTLS.ps1` because changes are not applied immediately
4) Exclude `Cached Docker images` section from docs if no containers are cached

## Check list
- [ ] Related issue / work item is attached
- [ ] Tests are written (if applicable)
- [ ] Documentation is updated (if applicable)
- [ ] Changes are tested and related VM images are successfully generated

### CODE REVIEW

**Review by al-cheb — DISMISSED** [review]

**Review by MaksimZhukov — APPROVED** [review]

**Review by AlenaSviridenko — DISMISSED** [review]

File: images/win/scripts/Installers/Install-Msys2.ps1 [hunk 1]
```diff
@@ -10,19 +10,68 @@
...
+
+  $pkgs = pacman.exe -Q
+
+  foreach ($arch in $archs)
+  {
+    Write-Host "`n$dash Installed $arch packages"
+    $pkgs | grep ^${arch}-
+  }
+}
+
```
Comment (miketimofeev) [inline images/win/scripts/Installers/Install-Msys2.ps1:72]: extra line

File: images/win/scripts/Installers/Install-Msys2.ps1 [hunk 2]
```diff
@@ -35,40 +84,10 @@ pacman.exe -Syuu  --noconfirm
 taskkill /f /fi "MODULES eq msys-2.0.dll"
```
Comment (miketimofeev) [inline images/win/scripts/Installers/Install-Msys2.ps1:84]: Do we want to do all these steps in case we don't install any packages?

Comment (maxim-lobanov) [reply images/win/scripts/Installers/Install-Msys2.ps1:84]: I am not sure about exact purpose of these lines so decided to keep them. Do you have any context?
May be @al-cheb knows

Comment (maxim-lobanov) [reply images/win/scripts/Installers/Install-Msys2.ps1:84]: Per discussion with @al-cheb , these lines should be run even if we don't install packages.

**Review by al-cheb — APPROVED** [review]

**Review by miketimofeev — APPROVED** [review]

**Pull request merged**